### PR TITLE
Add support to go modules

### PR DESCRIPTION
**Pull request by arthur29**

### CODE REVIEW

File: mongo/interfaces/interfaces.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type MongoDB interface {
 type Collection interface {
 	Find(query interface{}) Query
 	FindId(id interface{}) Query
+	Pipe(pipeline interface{}) Pipe
```
Comment (gabrielcorado) [inline mongo/interfaces/interfaces.go]: Why was this function added? Did we update the mongo lib version?

Comment (arthur29) [reply mongo/interfaces/interfaces.go]: No, sorry it is fruit of a bad rebase. It is fixed!

File: testing/log.go [hunk 2]
```diff
@@ -41,7 +41,7 @@ type containLogMessageMatcher struct {
 }
 
 func (matcher *containLogMessageMatcher) Match(actual interface{}) (success bool, err error) {
-	entries := actual.([]*logrus.Entry)
+	entries := actual.([]logrus.Entry)
```
Comment (gabrielcorado) [inline testing/log.go:44]: Is this change required?

File: redis/redis.go [hunk 3]
```diff
@@ -27,11 +27,11 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/bsm/redis-lock"
+	lock "github.com/bsm/redis-lock"
```
Comment (gabrielcorado) [inline redis/redis.go:30]: I guess this change is to make the name more consistent, right?

File: kafka/kafka_consumer_test.go [hunk 4]
```diff
@@ -366,7 +366,7 @@ var _ = Describe("Kafka Extension", func() {
 				defer client.StopConsuming()
 				go client.ConsumeLoop()
 
-				Eventually(func() []*logrus.Entry {
+				Eventually(func() []logrus.Entry {
```
Comment (gabrielcorado) [inline kafka/kafka_consumer_test.go:369]: Is this change required?

File: go.mod [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	gopkg.in/inf.v0 v0.9.0 // indirect
+	gopkg.in/ini.v1 v1.62.0 // indirect
+	gopkg.in/mgo.v2 v2.0.0-20160818020120-3f83fa500528
+	gopkg.in/mgutz/dat.v2 v2.0.0-20171004160617-d76e4f81c4ef
+	gopkg.in/olivere/elastic.v5 v5.0.66
+	gopkg.in/stretchr/testify.v1 v1.2.2 // indirect
+	mellium.im/sasl v0.2.1 // indirect
+)
+
+replace github.com/codahale/hdrhistogram => github.com/HdrHistogram/hdrhistogram-go v0.0.0-20200919145931-8dac23c8dac1
```
Comment (gabrielcorado) [inline go.mod:95]: Is this replace for our usage? If yes, why not change it?

**Review by gabrielcorado — APPROVED** [review]
LGTM. I just left a few comments but nothing blocking.

File: testing/log.go (hunk 2 above)
Comment (arthur29) [reply testing/log.go:44]: Yes, it breaks the tests because one of the updates was the logrus

File: kafka/kafka_consumer_test.go (hunk 4 above)
Comment (arthur29) [reply kafka/kafka_consumer_test.go:369]: Yes, it breaks the tests because one of the updates was the logrus

File: go.mod (hunk 5 above)
Comment (arthur29) [reply go.mod:95]: This replace is for the jaeger dependecy

File: redis/redis.go (hunk 3 above)
Comment (arthur29) [reply redis/redis.go:30]: Yes, it is right!
